control/derivative.go

```diff
@@ -90,6 +90,9 @@ func (d *derivative) Next(ctx context.Context, x []*Signal, dt time.Duration) ([
 	if d.stencil.Type == "backward" {
 		for idx, s := range x {
 			d.px[idx] = append(d.px[idx][1:], s.GetSignalValueAt(0))
+			if math.IsNaN(d.px[idx][0]) {
+				continue
+			}
```

Comment:
okay John helped me debug, we initialized all the derivative values to NaN and later skip any iterations in which the value is NaN. what was happening before was the position from the endpoint was a non-zero number, and the first endpoint number was 0 resulting in a large value passed into sum, which caused a large negative error. now we don't send anything to sum until it's a change since the last position is the actual last position and not 0